### ISSUE TITLE
Possible bad switch in optimizer

### ISSUE THREAD

**Issue by loliverhennigh** [rooa/eve#1]
Hey, I think there is an issue here in the code where the switch statement. It checks if t is greater the 1 however t is 1. because there is `t = self.iterations + 1` above. For me this was returning d_t on the first step and causing the loss to diverge instantly. I set the value to 2 and everything works now. I am using this optimizer with tensorflow and not Keras so maybe that is messing something up too...

https://github.com/rooa/eve/blob/master/eve/optim/eve.py#L67

**Comment by jayanthkoushik** [comment]
Hi, `self.iterations` is initialized to 0 (https://github.com/rooa/eve/blob/master/eve/optim/eve.py#L27). So `t` should be 1 on the first step. Do you mind sharing how you are using the code?
